controller/controlplane/controller.go

```diff
@@ -535,82 +365,73 @@ func (r *Reconciler) patchStatus(ctx context.Context, logger logr.Logger, update
-		}
-		return "", res, nil // requeue will be triggered by the creation or update of the owned object
+
+	subject := fmt.Sprintf("%s.%s", cp.Name, cp.Namespace)
+	template := x509.CertificateRequest{
+		Subject: pkix.Name{
+			CommonName:   subject,
+			Organization: []string{"Kong, Inc."},
+			Country:      []string{"US"},
+		},
+		SignatureAlgorithm: signatureAlgorithm,
+		DNSNames:           []string{subject},
 	}
 
-	log.Trace(logger, "ensuring admission webhook certificate")
-	res, admissionWebhookCertificateSecret, err := r.ensureAdmissionWebhookCertificateSecret(ctx, logger, cp, admissionWebhookService)
+	caCertBlock, _ := pem.Decode(caSecret.Data["tls.crt"])
+	if caCertBlock == nil {
+		return nil, nil, errors.New("failed to decode CA certificate")
+	}
+	caCert, err := x509.ParseCertificate(caCertBlock.Bytes)
 	if err != nil {
-		return "", res, err
+		return nil, nil, fmt.Errorf("failed to parse CA certificate: %w", err)
 	}
-	if res != op.Noop {
-		if !webhookEnabled {
-			log.Debug(logger, "admission webhook service certificate has been removed")
-		} else {
-			log.Debug(logger, "admission webhook service certificate has been created/updated")
-		}
-		return "", res, nil // requeue will be triggered by the creation or update of the owned object
+	caKeyBlock, _ := pem.Decode(caSecret.Data["tls.key"])
+	if caKeyBlock == nil {
+		return nil, nil, errors.New("failed to decode CA key")
 	}
 
-	log.Trace(logger, "ensuring admission webhook configuration")
-	res, err = r.ensureValidatingWebhookConfiguration(ctx, cp, admissionWebhookCertificateSecret, admissionWebhookService, enforceConfig)
+	caSigner, signatureAlgorithm, err := secrets.ParsePrivateKey(caKeyBlock)
 	if err != nil {
-		return "", res, err
-	}
-	if res != op.Noop {
-		if !webhookEnabled {
-			log.Debug(logger, "ValidatingWebhookConfiguration has been removed")
-		} else {
-			log.Debug(logger, "ValidatingWebhookConfiguration has been created/updated")
-		}
+		return nil, nil, fmt.Errorf("failed to parse CA key: %w", err)
 	}
-	if webhookEnabled {
-		return admissionWebhookCertificateSecret.Name, res, nil
-	}
-	return "", res, nil
-}
 
-func isAdmissionWebhookEnabled(ctx context.Context, cl client.Client, logger logr.Logger, cp *operatorv1beta1.ControlPlane) bool {
-	if cp.Spec.Deployment.PodTemplateSpec == nil {
-		return false
+	der, err := x509.CreateCertificateRequest(rand.Reader, &template, priv)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to create certificate request: %w", err)
 	}
 
-	container := k8sutils.GetPodContainerByName(&cp.Spec.Deployment.PodTemplateSpec.Spec, consts.ControlPlaneControllerContainerName)
-	if container == nil {
-		return false
+	policy := &config.Signing{
+		Default: &config.SigningProfile{
+			Usage: []string{
+				string(certificatesv1.UsageKeyEncipherment),
+				string(certificatesv1.UsageDigitalSignature),
+				string(certificatesv1.UsageClientAuth),
+			},
+			Expiry: time.Hour * 24 * 365,
```

Comment:
Do we have an issue for handling expiration of these certificates? If not let's create a tracking issue for that.

Reply:
Here you are https://github.com/Kong/gateway-operator/issues/1666